src/main/java/edu/ucsd/workflow/RemoveRedundantTSVRows.java

```diff
@@ -0,0 +1,220 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+
+public class RemoveRedundantTSVRows
+{
+	/*========================================================================
+	 * Constants
+	 *========================================================================*/
+	private static final String USAGE = "java -jar CCMSWorkflowUtils.jar" +
+		"\n\t-input <InputFile> (TSV file with header)" +
+		"\n\t-output <OutputFile> (Input file with redundant rows removed)" +
+		"\n\t-columns <ColumnHeaders> (Comma-separated list of columns " +
+		"for which only the first row per unique combination of these " +
+		"columns should be retained)";
+
+	/*========================================================================
+	 * Public interface methods
+	 *========================================================================*/
+	public static void main(String[] args) {
+		RemoveRedundantTSVRowsOperation removeRedundantTSVRows = extractArguments(args);
+		if (removeRedundantTSVRows == null)
+			die(USAGE);
+		// remove redundant rows in the TSV
+		BufferedReader input = null;
+		PrintWriter output = null;
+		Throwable thrownError = null;
+		int linesInInputFile = 1;
+		int linesInOutputFile = 1;
+		try {
+			input = new BufferedReader(new FileReader(
+				removeRedundantTSVRows.inputFile));
+			// set up output file writer
+			output = new PrintWriter(removeRedundantTSVRows.outputFile);
+			System.out.println(String.format(
+				"Writing the non-redundant rows in file \"%s\" to output file \"%s\".",
+				removeRedundantTSVRows.inputFile,
+				removeRedundantTSVRows.outputFile));
+			// read header
+			String headerStr = input.readLine();
+			String[] header = headerStr.trim().split("\t");
+			String columnsStr = "";
+			Set<List<String>> uniqueLineIDs = new HashSet<>();
+			for (int i=0; i<header.length; i++) {
+				header[i] = header[i].trim();
+			}
+			for (int i=0; i<removeRedundantTSVRows.columns.length; i++) {
+				String column = removeRedundantTSVRows.columns[i];
+				if (!(Arrays.asList(header).contains(column))) {
+					throw new IllegalArgumentException(
+						String.format("Column \"%s\" was not found in input header.",
+						column));
+				}
+				columnsStr += column + ", ";
+			}
+			if (columnsStr.endsWith(", "))
+				columnsStr = columnsStr.substring(0, columnsStr.length() - ", ".length());
+			System.out.println(String.format(
+				"Retaining the first row in input file per unique combination of " +
+				"the following columns: %s", columnsStr));
+			output.println(headerStr);
+			// read the remaining lines, and write the non-redundant ones to the output file
+			String lineStr = null;
+			while ((lineStr = input.readLine()) != null) {
+				linesInInputFile++;
+				String[] line = lineStr.trim().split("\t");
+				List<String> uniqueLineID = new ArrayList<>();
+				for (int i=0; i<line.length; i++) {
+					if (Arrays.asList(removeRedundantTSVRows.columns).contains(
```

Comment:
Same objection as above. This code will scale poorly to large files since you're repeating the same static operation on every line.